### ISSUE TITLE
Font UbuntuMono32.pf2 in grub2-theme-mint-2k not signed?

### ISSUE THREAD

**Issue by kochinc** [linuxmint/linuxmint#679]
I installed the grub2-theme-mint-2k package on my Linux Mint and found no changes on Grub menu. Searched Google and got  [this post](https://forums.linuxmint.com/viewtopic.php?t=411725) reporting possibly the same issue. The OP found commenting out the GRUB_FONT line allows the theme to work. However, that may also cause the theme to use the default font, which is smalled than intended.

On booting up, I dropped into the command line of Grub and used loadfont command to load the font. That gave me a message saying the secure boot policy doesn't allow it (paraphrasing). Can it be that the font UbuntuMono32.pf2 is not signed in Grub?

Thanks in advance.

**Comment by silent-thoughts** [comment]
Yes, I can confirm this is exactly the issue. With the help of Claude, I have put this together:

Title: grub2-theme-mint-2k font and background blocked by Secure Boot on HiDPI (2K) display

System:
- Dell XPS 13 9320 Plus
- Display: 1920x1200, 13.4" panel (~172 PPI, HiDPI)
- Linux Mint 22.3 Cinnamon
- GRUB 2.12
- Secure Boot: Enabled (Deployed Mode, Microsoft UEFI CA enabled)
- BitLocker with TPM+PIN also active on the Windows 11 side of this dual-boot machine

Problem:
On a 172 PPI display, the default GRUB font (UbuntuMono32.pf2, shipped by
grub2-theme-mint-2k) renders too small to comfortably read. Installing
grub2-theme-mint-2k is the documented fix for HiDPI screens, but with
Secure Boot enabled, the package's own font file fails to load and the
theme's background image also fails to load — while the theme's logo/icon
pixmaps (Mint logo, Windows logo, selection frame) load and render fine.

Steps taken:
1. Installed grub2-theme-mint-2k (1.2.3), confirmed via `dpkg -L` that all
   expected files (background.png, icons, theme.txt, UbuntuMono32.pf2,
   the /etc/default/grub.d/63_mint-theme-2k.cfg conffile) were present after
   a forced reinstall (`apt install --reinstall -o Dpkg::Options::="--force-confmiss"`,
   matching Mint's own documentation for HiDPI setups).
2. Built a custom 60pt font (grub-mkfont from Ubuntu-R.ttf) and pointed
   GRUB_FONT at it via a separate /etc/default/grub.d/99-custom-grub-font.cfg
   to avoid conflicting with the package's own config. Confirmed via
   `grub.cfg` that `loadfont` correctly targeted the custom file, with no
   competing loadfont calls. Rendered text size was unchanged from the
   default — the custom font never actually took visual effect.
3. Added a debug menuentry to /etc/grub.d/40_custom that explicitly calls
   `background_image` on the theme's background.png. This produced:
     error: file '/boot/grub/themes/linuxmint-2k/background.png' not found.
     error: prohibited by secure boot policy.
   (Note: our own boot partition is separate from root, so the correct
   in-GRUB path is /grub/themes/... rather than /boot/grub/themes/...; once
   corrected, the command reports "success" but the background still does
   not render, even with gfxterm confirmed active via `terminal_output`.)
4. Disabled Secure Boot entirely in BIOS as an isolation test. Result: the
   theme's logo/frame remained visible, and GRUB fell back to its built-in
   font (unicode.pf2) instead of the package's UbuntuMono32.pf2 — producing
   large, readable text, but still no background image, even with Secure
   Boot off. Re-enabling Secure Boot reverted text to the small default size.
5. Fully purged (`apt purge`) and cleanly reinstalled the theme package from
   scratch (confirmed byte-identical files via file listing) — background
   still did not render, ruling out file corruption or a bad install as the
   cause.

Conclusion:
- The package and its files are not at fault — verified via clean reinstall.
- Secure Boot (likely via the shim/TPM measured-boot chain rather than
  Secure Boot's signature check alone, since disabling Secure Boot in BIOS
  did not fully restore normal behavior) blocks loading of the theme's font
  and background image specifically, while still permitting the smaller
  icon/logo pixmaps to load via a different code path.
- This leaves HiDPI users who also want Secure Boot enabled (a common and
  reasonable combination, especially alongside Windows BitLocker on
  dual-boot machines) with no way to get a readable GRUB font size.

Suggestion:
It would help HiDPI users a great deal if grub2-theme-mint-2k shipped a
Secure-Boot-signed font (or if the package's install/postinst script
detected Secure Boot status and warned the user that the custom font will
silently fail to load under Secure Boot). Alternatively, documentation
noting this limitation would save other users the many hours we spent
diagnosing it.

Happy to provide further logs or test additional configurations if useful.

Tomas Risberg

**Comment by silent-thoughts** [comment]
Follow-up: the background image issue seems to have resolved itself after
toggling Secure Boot off and back on in BIOS — no relevant package or
firmware updates (checked `fwupdmgr get-history`: none) occurred in
between, so it's unclear exactly what changed. Possibly a stale Secure
Boot key database that got rebuilt by the toggle. The background and
theme icons now render correctly with Secure Boot enabled.

The font size issue remains unresolved, though: GRUB_FONT still doesn't
appear to take effect (neither the package's own UbuntuMono32.pf2 nor a
custom-built larger font), so HiDPI displays are still stuck with
small/default-sized text. Leaving this here in case the background
episode helps someone else who hits the same transient state — but the
font sizing problem for HiDPI users is still open as far as I can tell.

Tomas Risberg
